### DISCUSSION TITLE
Not Authenticating login in the terminal

### DISCUSSION THREAD

**Discussion by tylerlang94** [community/community#22445] [New to GitHub]
<p>So this is my first time trying to commit so this very well may be user error but I was unable to find anything through my searching. When I try to make my first push by pushing the origin to github I get a login window. When I type my username it changes it to <a href="mailto:user.name@github.com">user.name@github.com</a> and says that they were unable to authenticate. It is pulling this error:<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/6/9/69fee6084eb35cef93555869f721111ffe1423da.png" data-download-href="/uploads/short-url/f7GfCT98C9LzUe4GqOKjf6z8UmS.png?dl=1" title="giterror" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/92273345/181070950-a6ee1641-1ee5-4edb-bf0d-6c9f06f66044.png" alt="giterror" data-base62-sha1="f7GfCT98C9LzUe4GqOKjf6z8UmS" width="690" height="84" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/6/9/69fee6084eb35cef93555869f721111ffe1423da_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">giterror</span><span class="informations">903×110 3.66 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>I tried using a token to login and when I set that up I set it to be open forever with all permissions. That worked for the first push; however, I am still getting a login window everytime I want to push to my repository. Do I need to make a token everytime? Anyone that can help this newbie is much appreciated!</p>
<p>EDIT:<br>
I set up caching credentials in hopes that this would work and now I am getting permission denied on everything (git add, git commit, and git push) unless I run it as a sudo command. As long as I am running everything as root it works and doesn’t ask for credentials again.</p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="tylerlang94" data-post="1" data-topic="206043">
<div class="title">
<div class="quote-controls"></div>
 tylerlang94:</div>
<blockquote>
<p>As long as I am running everything as root it works and doesn’t ask for credentials again.</p>
</blockquote>
</aside>
<p>Please do not run Git as root. Unless the files it manages are supposed to be owned by root, Git does not do anything that requires root access, and for security you should never provide root access if it isn’t needed. In fact running Git as root is likely to <em>cause</em> problems, because newly created files will then be owned by root and not writable for your normal user. :warning:</p>
<p>It’s hard to say what might be causing your problems without more details about your setup, the error messages in your screenshot look like you might have previously stored credentials in Kwallet (KDE). Maybe you just need to update those with a token? Or you changed your desktop environment and now Kwallet isn’t available any more?</p>
<p>Finally, you don’t need a token with all permissions. For pushing/fetching <strong>repo</strong> and <strong>workflow</strong> should be sufficient.</p>

**Comment by tylerlang94** [comment]
<p>Thanks for the information! I didn’t think of any of those possible issues when running as root. I haven’t changed my DE from KDE since I started using git. I will check my kwallet to see if my original user.name and password credentials were saved in there when I get home and hopefully that fixes my issue.</p>

**Comment by tylerlang94** [comment]
<p><a class="mention" href="/airtower-luna">@airtower-luna</a> Sorry for the late response but I was just now able to test. You were correct with the problem being in KDEwallet. I created a new token with the permissions that you suggested and made sure it was in KDEwallet correctly. Was able to do multiple pushes without it asking for key again. Thank you for the assistance!</p>
